webapp/src/utils/regex_utils.test.js

```diff
@@ -0,0 +1,18 @@
+import {validateGitlabURL} from './regex_utils';
+
+describe('validateGitlabURL should work as expected', () => {
+    it('Should return true for valid GitLab repository URL', () => {
+        const URL = 'https://gitlab.com/username/repo/-/merge_requests/1234';
+        expect(validateGitlabURL(URL)).toBe(true);
+    });
+
+    it('Should return false for invalid GitLab repository URL', () => {
+        const URL = 'https://github.com/username/repo';
+        expect(validateGitlabURL(URL)).toBe(false);
+    });
+
+    it('Should return false for non-URL string input', () => {
+        const URL = 'not a URL';
+        expect(validateGitlabURL(URL)).toBe(false);
+    });
+});
```

Comment:
Can we test for self-hosted GitLab instances as well? Ideally we can test using these instances in the redux store, and test the normal flow of checking the store for the connected URLs

Reply:
@mickmister Can you please elaborate what do you mean by `Ideally we can test using these instances in the redux store, and test the normal flow of checking the store for the connected URLs` ?

Reply:
@Nityanand13 Is the connected GitLab instance's URL in the redux store? If not, we should make sure the client has access to that, and make it so the test covers the path where we use the store to determine what GitLab URL to look for.

Reply:
@mickmister It is always stored in the redux store

Reply:
@mickmister Just to make sure there is not any confusion I want to clear the flow of how we are checking the URL. Firstly in the if statement, we check if the URL is related to the connected instance or not then `validateGitlabURL` checks if the URL is valid (like the last part of the URL is a number or not (merge_request id) and similar checks like this). Then finally we extract the parts of the URL with `getInfoAboutLink` and make API requests according to the type of the link.

So, there is no use of the connected GitLab instance URL stored in the redux store inside the `validateGitlabURL` function. So, as there is no use of the connected instance URL inside this function we cannot actually test it.

Reply:
> So, there is no use of the connected GitLab instance URL stored in the redux store inside the validateGitlabURL function. So, as there is no use of the connected instance URL inside this function we cannot actually test it.

This makes sense to me. It seems all we are doing is extracting the connected gitlab URL from the redux store and doing the `url.origin` comparison, with no other logic to test there.